src/main/java/com/endava/web/StudentController.java

```diff
@@ -0,0 +1,43 @@
+package com.endava.web;
+
+import com.endava.repository.Student;
+import com.endava.service.StudentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/student")
+public class StudentController {
+
+    private StudentService studentService;
+
+    @Autowired
+    public StudentController(StudentService studentService) {
+        this.studentService = studentService;
+    }
+
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+    public Student findStudentById(@RequestParam int id) {
+        if (studentService.findStudentById(id) == null) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Student with id=" + id + " doesn't exist.");
+        }
+        return studentService.findStudentById(id);
+    }
+
+    @RequestMapping(method = RequestMethod.POST, consumes = "application/json")
+    @ResponseBody
+    public void insertStudent(@RequestBody Student student) {
```

Comment:
what happens if student is null

Reply:
If client asks for student with id=12340000 and that id doesn't exist, 
then client will receive 404 NOT FOUND status and message "Student with id=12340000 doesn't exist."

Reply:
And if client sends null object instead of student object

Reply:
Aha, tnx.

For empty body: status: 415 Unsupported Media Type
For no specified name and id : new Student is created (name:null, id:0)

After corection:
For no specified name and id : status: 400 Bad Request